README.md

```diff
@@ -1,53 +1,109 @@
-# Wiki
+# Publish to GitHub wiki action
 
-This action takes the contents of a provided directory and synchronizes it with your
-repository's wiki. This allows, for example, pull requests for the wiki.
+📖 Deploy docs from your source tree to the GitHub wiki
 
-The goal is to expose the wiki to PRs, so that potential contributors who aren't
-collaborators can contribute to the wiki and go through a review process. Alternatively,
-one can disabled the "Restrict editing to collaborators only" setting for their wiki,
-but this allows the community to make unreviewed, potentially unwanted, changes.
+<div align="center">
 
-By default, it will use the `.github/wiki` directory.
+![](https://placekitten.com/600/400)
 
-## Before using this action
+<!-- prettier-ignore -->
+[Awesome GitHub wikis](https://github.com/MyHoneyBadger/awesome-github-wiki#readme)
+| [🆕 What's new in v1](#tbd)
+| [GitHub workflow permissions docs](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs)
 
-You probably want to commit your wiki in its initial state before using this action.
-This is because this action will commit *all* differences between the wiki and the
-`.github/wiki` (or other specified directory) contents.
+</div>
 
-```bash
-git clone https://github.com/octocat/my-repo.wiki.git .github/wiki
-rm -rf .github/wiki/.git
-git add .github/wiki
-git commit -m "Add wiki to repo"
-```
+🌐 Works across repositories (with a [PAT]) \
+📚 Pretty interface for Markdown docs \
+⤴️ Lets you open PRs for wiki docs
 
 ## Usage
 
-Unfortunately, `${{ github.token }}` does not have access to updating wiki
-repositories, so it is necessary to create a new PAT.
+❗ Make sure you turn on the GitHub wiki feature in your repo's settings menu.
+You'll also need to _manually_ create a dummy page to initialize the git repo
+that powers the GitHub wiki. If you don't, when we push to `your-repo.wiki.git`,
+your workflow will fail because the wiki doesn't exist.
+
+Create a workflow file named something like `.github/workflows/wiki.yml` and add
+a job that uses this action to it!
+
+```yml
+name: Wiki
+on:
+  push:
+    branches: [main]
+    paths: [wiki/**, .github/workflows/wiki.yml]
+concurrency:
+  group: wiki
+  cancel-in-progress: true
+permissions:
+  content: write
+jobs:
+  wiki:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: spenserblack/actions-wiki@v0.1.1
+```
```

Comment:
I think we can just link to our `update-wiki` workflow here, instead of duplicating the code. This also lets readers see that we're "dogfooding".

Reply:
In this case, duplication I think is justified to improve the UX of getting started. You don't want to say "yeah just copy us!" since there's _no surrounding context_. Having it embedded on the same page as the surrounding options table, with some earnings about getting started needing an existing git wiki repo, and the other notes and information is vital to UX so you don't need to wait 3s every time you want to swap between the example code and the explanation of that code. It's easier if they're just a quick scroll away instead of a full page load.

Now... if we could _embed_ the file in the readme without code duplication, that'd be awesome! GitHub Blocks might do that in the future? https://blocks.githubnext.com/

**BUT** I do agree that we could and should link to a live example of us dogfooding the thing. Just not as the _only_ example.

Reply:
Good points. Swapping between example and explanation definitely convinced me :slightly_smiling_face:
I touched on this in the discussion, but if we also include some commented-code for the inputs, this could be pretty nice to use. Just copy+paste into your own workflow, and uncomment as needed.

One minor nitpick is that, instead of `actions-wiki@v0.1.1`, `actions-wiki@<commit-ish>`. I don't want to encourage a certain strategy, whether they want to lock to a certain version, use the latest major release, or use the bleeding edge version (`@main`).

Reply:
> One minor nitpick is that, instead of actions-wiki@v0.1.1, actions-wiki@<commit-ish>. I don't want to encourage a certain strategy, whether they want to lock to a certain version, use the latest major release, or use the bleeding edge version (@main).

I'd try to stick with an npm-like or pip-like default of either:
1. using the equivalent of `^v1.2.3` (so `@v1`)
2. using the equivalent of `==1.2.3` (so `@v1.2.3`)

commitish doesn't really work well because workflows _require the full 30-char commit id_ and don't accept a 5-char abbrev. sooooo it's a bit unwieldy imo.

Also the ecosystem has defaulted to recommending major versions like `actions/checkout@v3` in copy-paste examples, even outside of https://github.com/actions

i guess conclusion is that I think `@vN` is the best example thing to have, but since we don't have a `@v0` tag right now #21 , the next best one is `@v0.1.1` or whatever the latest release is.